File: src/Entity/Browser/Container/Tab/Page/Title.php
<?php

declare(strict_types=1);

namespace Yggverse\Yoda\Entity\Browser\Container\Tab\Page;

class Title
{
    public \GtkLabel $gtk;

    // Dependencies
    public \Yggverse\Yoda\Entity\Browser\Container\Tab\Page $page;

    // Defaults
    private int $_ellipsize = 3;
    private int $_length = 12;
    private string $_value = 'New page';

    public function __construct(
        \Yggverse\Yoda\Entity\Browser\Container\Tab\Page $page,
    ) {
        // Init dependencies
        $this->page = $page;

        // Init container
        $this->gtk = new \GtkLabel(
            $this->_value
        );

        $this->gtk->set_width_chars(
            $this->_length
        );

        $this->gtk->set_ellipsize(
            $this->_ellipsize
        );
    }

    public function setValue(
        ?string $value = null
    ): void
    {
        $this->gtk->set_text(
            is_null($value) ? $this->_value : trim(
                $value
            )
        );
    }
}

File: src/app/browser/header/bar/title.cpp
#include "title.h"

namespace app
{
    namespace browser
    {
        namespace header
        {
            namespace bar
            {
                // Construct
                Title::Title(
                    Bar *bar
                ) {
                    // Init dependencies
                    this->bar = bar;

                    // Init GTK
                    this->gtk = gtk_label_new(
                        Title::LABEL
                    );

                    // Render
                    gtk_widget_show(
                        GTK_WIDGET(
                            this->gtk
                        )
                    );
                }
            }
        }
    }
}
